### ISSUE TITLE
Build the Forge high-level API

### ISSUE THREAD

**Issue by NP-compete** [NP-compete/smithery#11]
## Problem

The README showcases a beautiful one-class API:

\`\`\`python
from smithery import Forge

forge = Forge.from_tools("./my_tools.json")
dataset = forge.generate_data(num_examples=5000)
model = forge.train(base_model="microsoft/Phi-3.5-mini-instruct", dataset=dataset)
report = forge.evaluate(model, dataset.test_split)
forge.export(model, format="gguf", quantize="q4_k_m")
\`\`\`

This \`Forge\` class doesn't exist yet. It's the glue that makes smithery feel like one tool instead of five separate modules.

## Proposed Solution

\`Forge\` is a **facade** that orchestrates the lower-level modules with sensible defaults:

\`\`\`python
class Forge:
    def __init__(self, tools: list[ToolDefinition]):
        self._tools = tools
    
    @classmethod
    def from_tools(cls, path: str) -> Forge:
        """Load tools from JSON, MCP config, or OpenAI format (auto-detected)."""
    
    @classmethod
    def from_mcp(cls, config_path: str) -> Forge:
        """Load tools from an MCP server config."""
    
    def generate_data(self, num_examples=5000, **kwargs) -> Dataset:
        """Generate synthetic training data. Wraps ToolCallGenerator."""
    
    def train(self, base_model="microsoft/Phi-3.5-mini-instruct",
              dataset=None, config=None, **kwargs) -> TrainedModel:
        """Fine-tune a model. Wraps AgentTrainer."""
    
    def evaluate(self, model, test_set, **kwargs) -> EvalMetrics:
        """Evaluate tool-calling accuracy. Wraps AgentBenchmark."""
    
    def export(self, model, format="gguf", **kwargs) -> Path:
        """Export model for deployment. Wraps export modules."""
    
    def forge(self, base_model="microsoft/Phi-3.5-mini-instruct",
              num_examples=5000, **kwargs) -> ForgeResult:
        """The one-shot method: generate → train → eval → export. All at once."""
\`\`\`

### The \`forge()\` method

This is the "I just want a model" button:
\`\`\`python
result = Forge.from_mcp("./server.json").forge()
# result.model_path, result.metrics, result.gguf_path
\`\`\`

Equivalent to running the full 4-step CLI pipeline.

### Auto-detection

\`from_tools(path)\` should auto-detect the format:
- \`.json\` file with \`"mcpServers"\` key → MCP config → use MCPImporter
- \`.json\` file with array of tool objects → raw tool definitions
- \`.json\` file with \`"functions"\` key → OpenAI format
- Directory path → tool descriptor files → use MCPImporter.from_tool_descriptors

## Acceptance Criteria

- [ ] \`Forge\` class exported from \`smithery.__init__\`
- [ ] \`from_tools()\` auto-detects input format (MCP config, JSON tools, OpenAI functions, directory)
- [ ] \`from_mcp()\` creates Forge from MCP server config
- [ ] \`generate_data()\` wraps ToolCallGenerator with sensible defaults
- [ ] \`train()\` wraps AgentTrainer, auto-selects recipe based on base_model
- [ ] \`evaluate()\` wraps AgentBenchmark, returns EvalMetrics
- [ ] \`export()\` wraps export modules, supports all formats
- [ ] \`forge()\` runs the full pipeline end-to-end
- [ ] Rich console output showing progress through each phase
- [ ] Unit tests for auto-detection logic
- [ ] Integration test for full \`forge()\` pipeline (tiny model)

## Dependencies

- Depends on #5 (data generation), #6 (MCP import), #7 (formats), #8 (training), #9 (eval), #10 (export)
- This is the capstone of Phase 1

## Estimated Effort

Small-Medium (2-3 days). This is primarily glue code, but the auto-detection and error handling need to be polished.
